File: example-codes-py/blackjack-and-a-deck-of-cards/hand.py
class Hand:
    def __init__(self, blackjack_card1, blackjack_card2):
        self.__cards = [blackjack_card1, blackjack_card2]

    def get_scores(self):
        totals = [0]

        for card in self.__cards:
            new_totals = []
            for score in totals:
                new_totals.add(card.face_value() + score)
                if card.face_value() == 1:
                    new_totals.add(11 + score)

            totals = new_totals

        return totals

    def add_card(self, card):
        self.__cards.add(card)

    # get highest score which is less than or equal to 21
    def resolve_score(self):
        scores = self.get_scores()
        best_score = 0
        for score in scores:
            if score <= 21 and score > best_score:
                best_score = score

        return best_score

File: example-codes/blackjack-and-a-deck-of-cards/java/Hand.java
package blackjack;

import java.util.ArrayList;
import java.util.List;

public class Hand {
    private List<BlackjackCard> cards;

    public Hand(BlackjackCard blackjackCard1, BlackjackCard blackjackCard2) {
        this.cards = new ArrayList<>();
        this.cards.add(blackjackCard1);
        this.cards.add(blackjackCard2);
    }

    public List<Integer> getScores() {
        List<Integer> totals = new ArrayList<>();
        totals.add(0);

        for (BlackjackCard card : cards) {
            List<Integer> newTotals = new ArrayList<>();
            for (int score : totals) {
                newTotals.add(card.getFaceValue() + score);
                if (card.getFaceValue() == 1) {
                    newTotals.add(11 + score);
                }
            }
            totals = newTotals;
        }

        return totals;
    }

    public void addCard(BlackjackCard card) {
        this.cards.add(card);
    }

    // Get highest score which is less than or equal to 21
    public int resolveScore() {
        List<Integer> scores = getScores();
        int bestScore = 0;
        for (int score : scores) {
            if (score <= 21 && score > bestScore) {
                bestScore = score;
            }
        }
        return bestScore;
    }

    public List<BlackjackCard> getCards() {
        return cards;
    }
}

